### PR TITLE
chore: add big list of naughty strings test for lexer

### PR DESCRIPTION
**Pull request by michaeljklein**
# Description

## Problem\*

Resolves https://github.com/noir-lang/noir/issues/4613

## Summary\*

Converts BLNS items into simple test programs and runs them through the lexer.

## Documentation\*

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[Exceptional Case]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: compiler/noirc_frontend/src/lexer/lexer.rs [hunk 1]
```diff
@@ -1087,4 +1087,201 @@ mod tests {
...
+
+    #[test]
+    fn test_big_list_of_naughty_strings() {
+        use std::io::Cursor;
+        use std::io::Read;
+        use std::mem::discriminant;
+        use std::path::Path;
+        use std::process::Command;
+
+        let blns_url = "https://raw.githubusercontent.com/minimaxir/big-list-of-naughty-strings/master/blns.base64.json";
```
Comment (TomAFrench) [inline compiler/noirc_frontend/src/lexer/lexer.rs:1210]: Let's just vendor this. The file is small enough that the cost of adding it to git is small relative to the overhead from downloading each time / relative complexity of caching.

Comment (TomAFrench) [reply compiler/noirc_frontend/src/lexer/lexer.rs:1210]: Having looked at the list, I think only a subset of it is really relevant to us anyway.

File: compiler/noirc_frontend/src/lexer/lexer.rs [hunk 2]
```diff
@@ -1087,4 +1087,211 @@ mod tests {
...
+                ",
+                        s
+                    ),
+                    // let s = r#"Simon says "hello world""#;
+                    format!(
+                        "
+                    let s = r#\"{}\"#;
+                ",
+                        s
+                    ),
```
Comment (jfecher) [inline compiler/noirc_frontend/src/lexer/lexer.rs:1156]: Some weird formatting going on here. Can it be fixed/condensed?

File: compiler/noirc_frontend/src/lexer/lexer.rs [hunk 3]
```diff
@@ -1087,4 +1087,211 @@ mod tests {
...
+    }
+
+    #[test]
+    fn test_big_list_of_naughty_strings() {
+        use std::mem::discriminant;
+
+        let blns_contents = include_str!(env!("BLNS_JSON_PATH"));
+        let blns_base64: Vec<String> =
+            serde_json::from_str(blns_contents).expect("BLNS json invalid");
+        blns_base64.into_iter().for_each(|blns_base64_str| {
```
Comment (jfecher) [inline compiler/noirc_frontend/src/lexer/lexer.rs:1248]:
```suggestion
        for blns_base64_str in blns_base64 {
```

File: compiler/noirc_frontend/src/lexer/lexer.rs [hunk 4]
```diff
@@ -1087,4 +1087,211 @@ mod tests {
...
+            blns_base64_to_statements(blns_base64_str).into_iter().for_each(
+                |(token_discriminator_opt, blns_program_strs)| {
```
Comment (jfecher) [inline compiler/noirc_frontend/src/lexer/lexer.rs:1250]:
```suggestion
            let statements = blns_base64_to_statements(blns_base64_str);
            for (token_discriminator_opt, blns_program_strs) in statements {
```

File: compiler/noirc_frontend/src/lexer/lexer.rs [hunk 5]
```diff
@@ -1087,4 +1087,211 @@ mod tests {
...
+                    blns_program_strs.into_iter().for_each(|blns_program_str| {
```
Comment (jfecher) [inline compiler/noirc_frontend/src/lexer/lexer.rs:1251]:
```suggestion
                for blns_program_str in blns_program_strs {
```

File: compiler/noirc_frontend/src/lexer/lexer.rs [hunk 6]
```diff
@@ -1087,4 +1087,211 @@ mod tests {
...
+                        }
+
+                        assert!(
+                            expected_token_found,
+                            "{}",
+                            format!(
+                                "expected token not found: {:?}\noutput:\n{:?}",
+                                token_discriminator_opt, result_tokens
+                            )
+                        );
```
Comment (jfecher) [inline compiler/noirc_frontend/src/lexer/lexer.rs:1291]: assert should already support format strings:
```suggestion
                        assert!(
                            expected_token_found,
                            "expected token not found: {token_discriminator_opt:?}\noutput:\n{result_tokens:?}",
                        );
```

**Review by jfecher — COMMENTED** [review]
Just some minor formatting items

File: compiler/noirc_frontend/src/lexer/lexer.rs (hunk 2 above)
Comment (michaeljklein) [reply compiler/noirc_frontend/src/lexer/lexer.rs]: yes, I used `{s}` instead and it's more condensed after `cargo fmt / clippy`

**Review by jfecher — COMMENTED** [review]
LGTM. Just some lexer test errors remaining

**Review by jfecher — APPROVED** [review]

**Pull request merged**